src/render/draw_custom.js

```diff
@@ -75,10 +76,25 @@ function drawCustom(painter: Painter, sourceCache: SourceCache, layer: CustomSty
 
         context.setDepthMode(depthMode);
 
+        const shaderProgram = implementation.getShaderProgram && implementation.getShaderProgram(painter.transform.getProjection());
         if (painter.transform.projection.name === "globe") {
-            implementation.render(context.gl, painter.transform.customLayerMatrix(), painter.transform.getProjection(), painter.transform.globeToMercatorMatrix(), globeToMercatorTransition(painter.transform.zoom));
+            if (shaderProgram) {
```

Comment:
If we're ok with the approach, I'll clean up this section for better error reporting as well as safeguard (ensuring that the user properly injected user code snippet helper).